include/zephyr/drivers/sensor_types.h

```diff
@@ -0,0 +1,312 @@
+	     "Too many sensor types");
+
+/**
+ * Header used in every structure containing batchable data from a sensor.
+ *
+ * The typical structure for sensor data looks like:
+ *
+ *   struct sensor_type_data {
+ *       struct sensor_data_header header;
+ *       struct sensor_type_sample_data {
+ *           uint32_t timestamp_delta;
+ *           union {
+ *               \<type\> value;
+ *               \<type\> interpretation0;
+ *               \<type\> interpretation1;
+ *           };
+ *       } readings[1];
+ *   };
+ *
+ * Despite 'readings' being declared as an array of 1 element,
+ * an instance of the struct will actually have 'readings' as
+ * an array of header.reading_count elements (which may be 1).
+ * The 'timestamp_delta' is in relation to the previous 'readings' (or
+ * the base_timestamp for readings[0].  So,
+ * Timestamp for readings[0] == header.base_timestamp +
+ *     readings[0].timestamp_delta.
+ * Timestamp for readings[1] == timestamp for readings[0] +
+ *     readings[1].timestamp_delta.
+ * And thus, in order to determine the timestamp for readings[N], it's
+ * necessary to process through all of the N-1 readings.  The advantage,
+ * though, is that our entire readings can span an arbitrary length of time,
+ * just as long as any two consecutive readings differ by no more than
+ * 4.295 seconds (timestamp_delta, like all time in the CHRE, is in
+ * nanoseconds).
+ *
+ * If a sensor has batched readings where two consecutive readings differ by
+ * more than 4.295 seconds, the CHRE will split them across multiple
+ * instances of the struct, and send multiple events.
+ *
+ * The value from the sensor is typically expressed in a union,
+ * allowing a generic access to the data ('value'), along with
+ * differently named access giving a more natural interpretation
+ * of the data for the specific sensor types which use this
+ * structure.  This allows, for example, barometer code to
+ * reference readings[N].pressure, and an ambient light sensor
+ * to reference readings[N].light, while both use the same
+ * structure.
+ */
+struct sensor_data_header {
+	/**
+	 * The base timestamp, in nanoseconds.
+	 */
+	uint64_t base_timestamp;
+
+	/**
+	 * The number elements in the 'readings' array.
+	 *
+	 * This must be at least 1.
+	 */
+	uint16_t reading_count;
+};
+
+/**
+ * Data for a sensor which reports on three axes.
+ *
+ * This is returned by:
+ * - Accelerometer readings and bias.
+ * - Gyroscope readings and bias.
+ * - Magnetometer readings and bias.
+ */
+struct sensor_three_axis_data {
+	/**
+	 * @see sensor_data_header
+	 */
+	struct sensor_data_header header;
+	struct sensor_three_axis_sample_data {
+		/**
+		 * @see sensor_data_header
+		 */
+		uint32_t timestamp_delta;
+		union {
+			fp_t values[3];
+			fp_t v[3];
+			struct {
+				fp_t x;
+				fp_t y;
+				fp_t z;
+			};
+			fp_t bias[3];
+			struct {
+				fp_t x_bias;
+				fp_t y_bias;
+				fp_t z_bias;
+			};
```

Comment:
I don't see the value in having so many ways to access the data. Isn't a bias just another instance of the same type of struct?

I think it would be sufficient to have:
```c
union {
	fp_t values[3]
	struct {
		fp_t x;
		fp_t y;
		fp_t z;
	};
};
```

Reply:
It's really just to have more verbose code. I don't feel strongly about it, but it's just a bit more readable since the struct just says that it has 3 dimensions, using `x_bias` is more obvious for what you're using it for. This is especially true once you start passing it around (and get further and further away from the API call that generated the data). If you feel strongly about it, I'm happy to reduce.

Reply:
I think it's overkill.

Reply:
Done